packages/credential-providers/README.md

```diff
@@ -172,24 +203,29 @@ const client = new FooClient({
   credentials: fromWebToken({
     // Required. ARN of the role that the caller is assuming.
     roleArn: "arn:aws:iam::1234567890:role/RoleA",
-    // Required. The OAuth 2.0 access token or OpenID Connect ID token that is provided by the
-    // identity provider.
+    // Required. The OAuth 2.0 access token or OpenID Connect ID token that is
+    // provided by the identity provider.
     webIdentityToken: await openIdProvider(),
-    // Optional. Custom STS client configurations overriding the default ones.
-    clientConfig: { region },
-    // Optional. A function that assumes a role with web identity and returns a promise fulfilled
-    // with credentials for the assumed role.
+    // Optional. A function that assumes a role with web identity and returns
+    // a promise fulfilled with credentials for the assumed role.
     roleAssumerWithWebIdentity,
     // Optional. An identifier for the assumed role session.
     roleSessionName: "session_123",
-    // Optional. The fully qualified host component of the domain name of the identity provider.
+    // Optional. The fully qualified host component of the domain name of the
+    // identity provider.
     providerId: "graph.facebook.com",
-    // Optional. ARNs of the IAM managed policies that you want to use as managed session.
+    // Optional. ARNs of the IAM managed policies that you want to use as
+    // managed session.
     policyArns: [{ arn: "arn:aws:iam::1234567890:policy/SomePolicy" }],
-    // Optional. An IAM policy in JSON format that you want to use as an inline session policy.
+    // Optional. An IAM policy in JSON format that you want to use as an
+    // inline session policy.
     policy: "JSON_STRING",
-    // Optional. The duration, in seconds, of the role session. Default to 3600.
+    // Optional. The duration, in seconds, of the role session. Default 3600.
     durationSeconds: 7200,
+    // Optional overrides. This is passed to an inner STS client
+    // instantiated to resolve the credentials. Region and profile
+    // are inherited from the upper client if present unless overridden.
```

Comment:
added terminology section to these docs. I'll continue using these terms since they appear in the code as well. The terminology sections explains outer/inner is as data/credentials. 

I can't always say data client since that can clash with data-plane control-plane client terminology.